src/overlays/actors/ovl_En_Horse_Normal/z_en_horse_normal.c

```diff
@@ -21,45 +48,635 @@ const ActorInit En_Horse_Normal_InitVars = {
+};
+
+void EnHorseNormal_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnHorseNormal* this = THIS;
+    s32 pad;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.gravity = -3.5f;
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawFunc_Squiggly, 20.0f);
+    this->actor.speedXZ = 0.0f;
+    this->actor.posRot2.pos = this->actor.posRot.pos;
+    this->actor.posRot2.pos.y += 70.0f;
+    this->type = 0;
+    this->animationIdx = 0;
+    Collider_InitCylinder(globalCtx, &this->bodyCollider);
+    Collider_SetCylinder(globalCtx, &this->bodyCollider, &this->actor, &sCylinderInit1);
+    Collider_InitJntSph(globalCtx, &this->headCollider);
+    Collider_SetJntSph(globalCtx, &this->headCollider, &this->actor, &sJntSphInit, this->headColliderItems);
+    Collider_InitCylinder(globalCtx, &this->cloneCollider);
+    Collider_SetCylinder(globalCtx, &this->cloneCollider, &this->actor, &sCylinderInit2);
+    func_80061ED4(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+    if (globalCtx->sceneNum == SCENE_SPOT20) {
+        if (this->actor.posRot.rot.z == 0 || gSaveContext.nightFlag) {
+            Actor_Kill(&this->actor);
+            return;
+        }
+        if (LINK_IS_CHILD) {
+            if (Flags_GetEventChkInf(0x14)) {
+                if (this->actor.posRot.rot.z != 3) {
+                    Actor_Kill(&this->actor);
+                    return;
+                }
+            } else if (this->actor.posRot.rot.z != 1) {
+                Actor_Kill(&this->actor);
+                return;
+            }
+        } else if (Flags_GetEventChkInf(0x18) || (DREG(1) != 0)) {
+            if (this->actor.posRot.rot.z != 7) {
+                Actor_Kill(&this->actor);
+                return;
+            }
+        } else if (this->actor.posRot.rot.z != 5) {
+            Actor_Kill(&this->actor);
+            return;
+        }
+        this->actor.shape.rot.z = 0;
+        this->actor.initPosRot.rot.z = this->actor.posRot.rot.z = this->actor.shape.rot.z;
+        func_800A663C(globalCtx, &this->skin, &D_06009FAC, &D_06004580);
+        SkelAnime_ChangeAnimDefaultStop(&this->skin.skelAnime, sAnimations[this->animationIdx]);
+        if ((this->actor.posRot.pos.x == -730.0f && this->actor.posRot.pos.y == 0.0f &&
+             this->actor.posRot.pos.z == -1100.0f) ||
+            (this->actor.posRot.pos.x == 880.0f && this->actor.posRot.pos.y == 0.0f &&
+             this->actor.posRot.pos.z == -1170.0f)) {
+            func_80A6C6B0(this);
+            return;
+        }
+    } else if (globalCtx->sceneNum == SCENE_MALON_STABLE) {
+        if (!gSaveContext.nightFlag) {
+            Actor_Kill(&this->actor);
+        } else {
+            func_800A663C(globalCtx, &this->skin, &D_06009FAC, &D_06004580);
+            SkelAnime_ChangeAnimDefaultStop(&this->skin.skelAnime, sAnimations[this->animationIdx]);
+            func_80A6C6B0(this);
+        }
+        return;
```

Comment:
```suggestion
            Actor_Kill(&this->actor);
            return;
        } else {
            func_800A663C(globalCtx, &this->skin, &D_06009FAC, &D_06004580);
            SkelAnime_ChangeAnimDefaultStop(&this->skin.skelAnime, sAnimations[this->animationIdx]);
            func_80A6C6B0(this);
            return;
        }
```
More readable imo than sharing the return statement